### ISSUE TITLE
java.lang.ClassCastException: org.apache.seatunnel.flink.sink.JdbcSink cannot be cast to org.apache.seatunnel.flink.batch.FlinkBatchSink[Bug] [Module Name] Bug title

### ISSUE THREAD

**Issue by libailin** [apache/seatunnel#1474]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

from hdfs to mysql error

### SeaTunnel Version

dev 

4be278ff73404159facb63c4f928583a803bfbcb

### SeaTunnel Config

```conf
env {
  execution.parallelism = 1
}

source {
  FileSource {
    path = "hdfs://namenode.xxx.net:9000/home/hadoop/hive/warehouse/test.db/test/p_day=20220119"
    format.type = "text"
    schema = "string"
    result_table_name = "test"
  }
}

transform {
  Sql {
    sql = "select * from test"
  }
}

sink {
  JdbcSink {
     driver = "com.mysql.jdbc.Driver"
     url = "jdbc:mysql://10.202.10.88:3306/seatunnel_test?useUnicode=true&characterEncoding=utf8&useSSL=false"
     username = "root"
     password = "123456"
     query = "insert into seatunnel_test(message) values(?)"
     batch_size = 1
  }
}
```

### Running Command

```shell
./bin/start-seatunnel-flink.sh -m yarn-cluster -ynm seatunnel --config ./config/flink_hive_mysql.conf
```

### Error Exception

```log
------------------------------------------------------------
 The program finished with the following exception:

org.apache.flink.client.program.ProgramInvocationException: The main method caused an error: org.apache.seatunnel.flink.sink.JdbcSink cannot be cast to org.apache.seatunnel.flink.batch.FlinkBatchSink
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:372)
        at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:222)
        at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:114)
        at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:812)
        at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:246)
        at org.apache.flink.client.cli.CliFrontend.parseAndRun(CliFrontend.java:1054)
        at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:1132)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1657)
        at org.apache.flink.runtime.security.contexts.HadoopSecurityContext.runSecured(HadoopSecurityContext.java:41)
        at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1132)
Caused by: java.lang.ClassCastException: org.apache.seatunnel.flink.sink.JdbcSink cannot be cast to org.apache.seatunnel.flink.batch.FlinkBatchSink
        at org.apache.seatunnel.flink.batch.FlinkBatchExecution.start(FlinkBatchExecution.java:72)
        at org.apache.seatunnel.Seatunnel.entryPoint(Seatunnel.java:103)
        at org.apache.seatunnel.Seatunnel.run(Seatunnel.java:61)
        at org.apache.seatunnel.SeatunnelFlink.main(SeatunnelFlink.java:29)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:355)
        ... 11 more
```

### Flink or Spark Version

flink 1.13.6

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by kalencaya** [comment]
already solved, please pull latest dev branch and try again

**Comment by ruanwenjun** [comment]
This should already been fixed by #1432

**Comment by CalvinKirs** [comment]
fixed
